posydon/unit_tests/utils/test_posydonwarning.py

```diff
@@ -0,0 +1,747 @@
+        assert len(warn1) == 1
+        # check output of FutureWarning
+        with warns(FutureWarning, match="Unit test") as warn2:
+            totest.Pwarn("Unit test", FutureWarning)
+        assert len(warn2) == 1
+        # check output of UserWarning (default if unspecified)
+        with warns(UserWarning, match="Unit test") as warn3:
+            totest.Pwarn("Unit test")
+        assert len(warn3) == 1
+
+    def test_SetPOSYDONWarnings(self):
+        totest.SetPOSYDONWarnings(action="once")
+        assert str(totest.warnings.filters[0]) == "('once', None, <class "+\
+               "'posydon.utils.posydonwarning.POSYDONWarning'>, None, 0)"
+        totest.SetPOSYDONWarnings()
+        assert str(totest.warnings.filters[0]) == "('default', None, <class "+\
+               "'posydon.utils.posydonwarning.POSYDONWarning'>, None, 0)"
+        # non POSYDON warnings have no effect
+        totest.SetPOSYDONWarnings(category=UserWarning)
+        assert str(totest.warnings.filters[0]) == "('default', None, <class "+\
+               "'posydon.utils.posydonwarning.POSYDONWarning'>, None, 0)"
+
+    def test_NoPOSYDONWarnings(self, reset_filter):
+        totest.NoPOSYDONWarnings()
+        assert str(totest.warnings.filters[0]) == "('ignore', None, <class "+\
+               "'posydon.utils.posydonwarning.POSYDONWarning'>, None, 0)"
+
+    def test_AllPOSYDONWarnings(self, reset_filter):
+        totest.AllPOSYDONWarnings()
+        assert str(totest.warnings.filters[0]) == "('always', None, <class "+\
+               "'posydon.utils.posydonwarning.POSYDONWarning'>, None, 0)"
+
+
+class TestPOSYDONWarning:
+    @fixture
+    def POSYDONWarning(self):
+        # initialize an instance of the class with defaults
+        return totest.POSYDONWarning()
+
+    # test the POSYDONWarning class
+    def test_init(self, POSYDONWarning):
+        assert isroutine(POSYDONWarning.__init__)
+        # check that the instance is of correct type and all code in the
+        # __init__ got executed: the elements are created and initialized
+        assert isinstance(POSYDONWarning, totest.POSYDONWarning)
+        assert POSYDONWarning.message == ''
+
+    def test_str(self, POSYDONWarning):
+        assert isroutine(POSYDONWarning.__str__)
+        assert str(POSYDONWarning) == "''"
+
+
+class TestApproximationWarning:
+    @fixture
+    def ApproximationWarning(self):
+        # initialize an instance of the class with defaults
+        return totest.ApproximationWarning()
+
+    # test the ApproximationWarning class
+    def test_init(self, ApproximationWarning):
+        assert isroutine(ApproximationWarning.__init__)
+        # check that the instance is of correct type and all code in the
+        # __init__ got executed: the elements are created and initialized
+        assert isinstance(ApproximationWarning, totest.ApproximationWarning)
+        assert ApproximationWarning.message == ''
+
+
+class TestBinaryParsingWarning:
```

Comment:
Can this be parameterised somehow?

It's a lot of the same code.

Reply:
To me it feels like we should keep it explicit. Otherwise, we get a test which covers more than one class.